sdk/core/core/inc/az_curl_adapter.h

```diff
@@ -5,17 +5,27 @@
 #define AZ_CURL_ADAPTER_H
 
 #include <az_callback.h>
+#include <az_http_request.h>
 #include <az_http_request_builder.h>
-#include <az_pair.h>
 #include <az_span_builder.h>
 #include <az_span_seq.h>
 #include <az_str.h>
 
-#include <_az_cfg_prefix.h>
-
 #include <curl/curl.h>
 #include <stdlib.h>
 
+#include <_az_cfg_prefix.h>
+
+// returning AZ error on CURL Error
+#define AZ_RETURN_IF_CURL_FAILED(exp) \
+  do { \
+    CURLcode const _result = (exp); \
+    if (_result != CURLE_OK) { \
+      fprintf(stderr, "curl error: %s\n", curl_easy_strerror(_result)); \
```

Comment:
remove `fprintf` :-)

Reply:
I need it until we have some reliable logging method

Reply:
I would suggest to open at least one GitHub issue and add a list of things that we have to fix. Later, we can split it to several issues. At least, we need to track it if it will be merged to master.